### PR TITLE
Add factor metadata value and new key

### PR DESCRIPTION
**Pull request by sonaisrayel**

### CODE REVIEW

File: core/factor.go [hunk 1]
```diff
@@ -63,20 +85,22 @@ func (factor *Factor) FromModel(mod *models.Factor) error {
...
+func (factor *FactorUpdateRequest) FromModel(mod *models.Factor) error {
+	factor.Publisher = mod.Publisher
+	factor.Domain = mod.Domain
+	factor.Device = mod.Device
+	factor.Factor = mod.Factor
+	factor.Country = mod.Country
 	return nil
 }
 
+func (factor *FactorUpdateRequest) ToRtRule() *FactorRealtimeRecord {
```
Comment (finkiel) [inline core/factor.go:97]: this function is not in use - you can delete it

File: core/factor.go [hunk 2]
```diff
@@ -63,20 +85,22 @@ func (factor *Factor) FromModel(mod *models.Factor) error {
...
-	for _, mod := range slice {
-		c := Factor{}
-		err := c.FromModel(mod)
-		if err != nil {
-			return eris.Cause(err)
-		}
-		*cs = append(*cs, &c)
-	}
-
+func (factor *FactorUpdateRequest) FromModel(mod *models.Factor) error {
```
Comment (finkiel) [inline core/factor.go:88]: I think that this function is not in use - please delete it if I'm right

File: core/factor.go [hunk 3]
```diff
@@ -128,38 +152,99 @@ func UpdateMetaData(c *fiber.Ctx, data *FactorUpdateRequest) error {
...
 		Device:    data.Device,
+		Factor:    data.Factor,
+		Country:   data.Country,
 	}
 
-	key := utils.CreateMetadataKey(metadataKey, "price:factor")
+	return modConf.Upsert(c.Context(), bcdb.DB(), true, []string{models.FactorColumns.Publisher, models.FactorColumns.Domain, models.FactorColumns.Device, models.FactorColumns.Country}, boil.Infer(), boil.Infer())
+}
+
+func (fs *FactorSlice) FromModelFactor(modFactors *models.Factor) {
```
Comment (finkiel) [inline core/factor.go:175]: same thing - not in use

File: core/confiant.go [hunk 4]
```diff
@@ -166,3 +165,66 @@ func UpdateConfiant(c *fiber.Ctx, data *ConfiantUpdateRequest) error {
...
+	finalOutput := struct {
+		Rules []ConfiantRealtimeRecord `json:"rules"`
+	}{Rules: finalRules}
+
+	value, err := json.Marshal(finalOutput)
+	if err != nil {
+		return eris.Wrap(err, "failed to marshal confiantRT to JSON")
+	}
+
+	key := utils.GetMetadataKey(updateRequest)
```
Comment (finkiel) [inline core/confiant.go:190]: not sure that I understand what you're doing here
the key value is exactly the same as updateRequest, why are you creating it?

if the function name is "getMetaDataKey" I expect it to return metadatakey (value) not an object

File: core/confiant.go [hunk 5]
```diff
@@ -166,3 +165,66 @@ func UpdateConfiant(c *fiber.Ctx, data *ConfiantUpdateRequest) error {
...
+	metadataKey := utils.CreateMetadataKey(key, "confiant:v2")
+	metadataValue := utils.CreateMetadataValue(updateRequest, metadataKey, value)
```
Comment (finkiel) [inline core/confiant.go:192]: you're not creating a metadataValue but metaDataObject (entire row) - the method name is incorrect

File: core/factor.go [hunk 6]
```diff
@@ -128,38 +152,99 @@ func UpdateMetaData(c *fiber.Ctx, data *FactorUpdateRequest) error {
...
+
+	value, err := json.Marshal(finalOutput)
 	if err != nil {
-		return utils.ErrorResponse(c, fiber.StatusInternalServerError, "Failed to insert metadata update to queue")
+		return eris.Wrap(err, "failed to marshal factorRT to JSON")
+	}
+
+	key := utils.GetMetadataKey(updateRequest)
+	metadataKey := utils.CreateMetadataKey(key, "price:factor:v2")
+	metadataValue := utils.CreateMetadataValue(updateRequest, metadataKey, value)
```
Comment (finkiel) [inline core/factor.go:208]: the name of the method is wrong
you're creating metadataObject/ Row not value

File: core/factor.go [hunk 7]
```diff
@@ -128,38 +152,99 @@ func UpdateMetaData(c *fiber.Ctx, data *FactorUpdateRequest) error {
...
+	}{Rules: finalRules}
+
+	value, err := json.Marshal(finalOutput)
 	if err != nil {
-		return utils.ErrorResponse(c, fiber.StatusInternalServerError, "Failed to insert metadata update to queue")
+		return eris.Wrap(err, "failed to marshal factorRT to JSON")
+	}
+
+	key := utils.GetMetadataKey(updateRequest)
+	metadataKey := utils.CreateMetadataKey(key, "price:factor:v2")
```
Comment (finkiel) [inline core/factor.go:207]: I think that all the values such as: price:factor:v2, confiant:v2.... should be declare as const

File: core/floor.go [hunk 8]
```diff
@@ -132,39 +140,89 @@ func UpdateFloorMetaData(c *fiber.Ctx, data *FloorUpdateRequest) error {
...
+
+	return modConf.Upsert(c.Context(), bcdb.DB(), true, []string{models.FloorColumns.Publisher, models.FloorColumns.Domain, models.FloorColumns.Device, models.FloorColumns.Country}, boil.Infer(), boil.Infer())
+}
+
+func SendFloorToRT(c context.Context, updateRequest FloorUpdateRequest) error {
+
+	modFloor, err := floorQuery(c, updateRequest)
+
+	if err != nil && err != sql.ErrNoRows {
+		return eris.Wrapf(err, "failed to fetch floors")
```
Comment (finkiel) [inline core/floor.go:168]: you can add the publisher name - "failed to fetch floors for publisher...."

File: core/floor.go [hunk 9]
```diff
@@ -132,39 +140,89 @@ func UpdateFloorMetaData(c *fiber.Ctx, data *FloorUpdateRequest) error {
...
-		Value:         []byte(floor),
+	value, err := json.Marshal(finalOutput)
+	if err != nil {
+		return eris.Wrap(err, "failed to marshal floorRT to JSON")
 	}
 
-	err = mod.Insert(c.Context(), bcdb.DB(), boil.Infer())
+	key := utils.GetMetadataKey(updateRequest)
+	metadataKey := utils.CreateMetadataKey(key, "price:floor:v2")
+	metadataValue := utils.CreateMetadataValue(updateRequest, metadataKey, value)
```
Comment (finkiel) [inline core/floor.go:186]: name of the method is incorrect

File: core/floor.go [hunk 10]
```diff
@@ -132,39 +140,89 @@ func UpdateFloorMetaData(c *fiber.Ctx, data *FloorUpdateRequest) error {
...
-		Key:           key,
-		TransactionID: bcguid.NewFromf(data.Publisher, data.Domain, time.Now()),
-		Value:         []byte(floor),
+	value, err := json.Marshal(finalOutput)
+	if err != nil {
+		return eris.Wrap(err, "failed to marshal floorRT to JSON")
 	}
 
-	err = mod.Insert(c.Context(), bcdb.DB(), boil.Infer())
+	key := utils.GetMetadataKey(updateRequest)
```
Comment (finkiel) [inline core/floor.go:184]: again, I don't understand why you are creating the key object

File: core/confiant.go (hunk 4 above)
Comment (sonaisrayel) [reply core/confiant.go:190]: I am using this function because I need to create Metadata interface

File: core/floor.go (hunk 10 above)
Comment (sonaisrayel) [reply core/floor.go:184]: I have this interface 
type UpdateRequest interface {
	GetPublisher() string
	GetDomain() string
	GetDevice() string
	GetCountry() string
}

and in order to have create metadakay I need to use this function

**Pull request closed without merging**